.github/workflows/macarm64.yml

```diff
@@ -0,0 +1,62 @@
+name: macOS arm64 CI
+on:
+  push:
+    branches:
+      - main
+      - '*.*.x'
+    tags:
+      - '*.*'
+      - '*.*.*'
+
+permissions: read-all
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  macos-arm64:
+    if: github.repository_owner == 'pyca'
+    runs-on: [self-hosted, macos, ARM64]
+    name: "macOS arm64"
+    timeout-minutes: 10
```

Comment:
This should have a matrix for python version, even if there's only one thing in it ATM.